client/layout/logged-out.jsx

```diff
@@ -161,7 +167,9 @@ const LayoutLoggedOut = ( {
 			<BodySectionCssClass group={ sectionGroup } section={ sectionName } bodyClass={ bodyClass } />
 			<div className="layout__header-section">
 				{ masterbar }
-				{ headerSection && <div className="layout__header-section-content">{ headerSection }</div> }
+				{ renderHeaderSection && (
+					<div className="layout__header-section-content">{ renderHeaderSection() }</div>
```

Comment:
I had to change renderHeaderSection to a function because of how setting locale works on the front end:
When setLocale is called, the client asynchronously requests a translation file for that locale.
When the XHR for that completes, a state change event occurs which triggers a re-render.
But if `headerSection` is passed in as a pre-rendered FC it was not being re-rendered with the new translations.